plutus-scb-client/src/MainFrame.purs

```diff
@@ -100,8 +117,51 @@ handleAction (ChainAction newFocus) = do
     (_chainState <<< _chainFocusAppearing)
     (zoomStateT _chainState $ Chain.handleAction newFocus mAnnotatedBlockchain)
 
-handleAction (InvokeContractEndpoint subaction) = do
-  liftEffect $ log $ "TRIGGER: " <> show subaction
+handleAction (ChangeContractEndpointCall contractInstanceId endpointIndex argumentIndex subaction) = do
+  modifying
+    ( _contractSignatures
+        <<< ix contractInstanceId
+        <<< _Success
+        <<< ix endpointIndex
+        <<< _arguments
+        <<< ix argumentIndex
+    )
+    (Schema.handleFormEvent initialValue subaction)
+
+handleAction (InvokeContractEndpoint contractInstanceId endpointForm) = do
+  let
+    endpointDescription :: EndpointDescription
+    endpointDescription = view (_schema <<< _FunctionSchema <<< _endpointDescription) endpointForm
+
+    encodedForm :: Warn (Text "It shouldn't be possible to get anything other than exactly 1 argument, but the code implies otherwise.") => Maybe RawJson
```

Comment:
I don't think it's a good idea to merge this PR with this in here

Reply:
Also, why? should `_arguments` be `_argument`? Not clear

Reply:
Fair. I'm going to address this in a separate PR, merge it, then rebase this one.